### PR TITLE
Add query_object_access and query_table_access models

### PR DESCRIPTION
**Pull request by ian-whitestone**
@NiallRees would appreciate your eyes on things like naming, the example query and documentation.

### CODE REVIEW

File: documentation/sample_queries.md [hunk 1]
```diff
@@ -54,6 +54,49 @@ where service in ('Storage')
 group by 1, 2, 3, 4
 ```
 
+### Unused Tables
+This query uses the `query_tables_accessed` model (available as of 1.6.0) to identify tables which have not been queried in the last 30 days. The total storage costs of each table is also shown, which leverages your Snowflake account's current storage rate (usually between $20-$40 per TB per month).
```
Comment (NiallRees) [inline documentation/sample_queries.md:58]:
```suggestion
This query uses the `query_tables_accessed` model (available as of 1.6.0) to identify tables which have not been queried in the last 30 days. The total storage costs of each table are also shown, which uses the account's current storage rate (usually between $20-$40 per TB per month).
``

File: models/query_tables_accessed.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (NiallRees) [inline models/query_tables_accessed.sql:1]: `query_history_table_access`?

Comment (NiallRees) [reply models/query_tables_accessed.sql:1]: or just `query_table_access` - past tense of access just sounds a bit odd imo

File: models/query_tables_accessed.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{{ config(
+    materialized='incremental',
+    unique_key=['_unique_key', 'query_start_time'],
+) }}
+
+with
+access_history as (
+    select *
+    from {{ ref('stg_access_history') }}
```
Comment (NiallRees) [inline models/query_tables_accessed.sql:10]: Should we get `OBJECTS_MODIFIED` while we're at it? Or do you think that should be a different table?

File: models/query_tables_accessed.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        split_part(objects_accessed.value:objectName::text, '.', 3) as table_name,
+        objects_accessed.value:columns as columns_array
+
+    from access_history, lateral flatten(access_history.direct_objects_accessed) as objects_accessed
+)
+
+select
+    *,
+    database_name || '.' || schema_name || '.' || table_name as full_table_name,
+    md5(concat(query_id, full_table_name)) as _unique_key
```
Comment (NiallRees) [inline models/query_tables_accessed.sql:35]: dbt practice is to put PK as first column - I'd also advocate for just calling it 'id' but I don't think we have a norm in this package yet so happy to go with `_unique_id` as the goto

File: models/query_tables_accessed.sql (hunk 3 above)
Comment (ian-whitestone) [reply models/query_tables_accessed.sql:10]: will have to be a different table

File: documentation/sample_queries.md [hunk 5]
```diff
@@ -54,6 +54,51 @@ where service in ('Storage')
 group by 1, 2, 3, 4
 ```
 
+### Unused Tables
+This query uses the `query_tables_accessed` model (available as of 1.6.0) to identify tables which have not been queried in the last 30 days. The total storage costs of each table are also shown, which uses the account's current storage rate (usually between $20-$40 per TB per month).
```
Comment (ian-whitestone) [inline documentation/sample_queries.md:58]:
```suggestion
This query uses the `query_table_access` model (available as of 1.6.0) to identify tables which have not been queried in the last 30 days. The total storage costs of each table are also shown, which uses the account's current storage rate (usually between $20-$40 per TB per month).
```

**Review by ian-whitestone — COMMENTED** [review]
@NiallRees can you have another look? I've decided two expand this to all objects accessed by a query (tables, secured views, tables from a data share, external tables) (`query_object_access` model) and then a separate view on top of `query_object_access`, `query_table_access`, which just filters down to tables managed by the account.

File: documentation/sample_queries.md [hunk 6]
```diff
@@ -54,6 +54,51 @@ where service in ('Storage')
 group by 1, 2, 3, 4
 ```
 
+### Unused Tables
```
Comment (jtalmi) [inline documentation/sample_queries.md:57]: note: you could include these as "analysis" queries in the analyses folder

Comment (ian-whitestone) [reply documentation/sample_queries.md:57]: we could, but planning to move these exapmles to a different site anyways. plus we'd lose the ability to have commentary mixed with code.

File: documentation/sample_queries.md [hunk 7]
```diff
@@ -54,6 +54,51 @@ where service in ('Storage')
...
+        total_storage_gb/1024 as total_storage_tb,
+        total_storage_tb*12*any_value(daily_rates.effective_rate) as annualized_storage_cost
+    from snowflake.account_usage.table_storage_metrics
+    inner join daily_rates
+        on daily_rates.is_latest_rate
+        and daily_rates.usage_type='storage'
+    where
+        not deleted
+    group by 1
+    order by annualized_storage_cost desc
```
Comment (NiallRees) [inline documentation/sample_queries.md:89]:
```suggestion
```

Unnecessary sort?

File: models/query_object_access.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
+version: 2
+
+models:
+  - name: query_object_access
+    description: This model tracks the objects directly accessed by a query. It contains 1 row per query per object. An object can be a regular table owned/created by your account, a table from a data share, a secured view, or an external table. See https://docs.snowflake.com/en/sql-reference/account-usage/access_history.html for more details. Access history is only available for Snowflake accounts on enterprise or higher, and therefore this model will be empty for standard accounts.
```
Comment (NiallRees) [inline models/query_object_access.yml:5]: How do we handle this so it doesn't error on standard accounts?

**Review by NiallRees — APPROVED** [review]

**Pull request merged**